src/cart/crt_hg.c

```diff
@@ -677,10 +730,10 @@ crt_get_info_string(bool primary, int provider, char **string, int ctx_idx)
 	}
 
 out:
-	if (*string == NULL)
+	if (rc == DER_SUCCESS && *string == NULL)
```

Comment:
I assume that we're the ones who did this, but it seems a bit odd to me that we're looking for a success return code and null pointer to then return a NOMEM return code. I'll separately look at crt_get_opx_info_string().

Reply:
Peace Charles S,
I was being overly cautious just ensuring that the result was in alignment with the string allocation.